tests/test_androidmodule_ctypes_finder.py

```diff
@@ -0,0 +1,124 @@
+
+import mock
+from mock import MagicMock
+import os
+import shutil
+import sys
+import tempfile
+
+sys.modules["jnius"] = MagicMock()  # fake jnius used later
+
+
+# Import the tested android._ctypes_library_finder module,
+# making sure android._android won't crash us!
+# (since android._android is android-only / not compilable on desktop)
+android_module_folder = os.path.abspath(os.path.join(
+    os.path.dirname(__file__),
+    "..", "pythonforandroid", "recipes", "android", "src"
+))
+sys.path.insert(0, android_module_folder)
+sys.modules['android._android'] = MagicMock()
+import android._ctypes_library_finder
+sys.path.remove(android_module_folder)
+
+
+def test_get_activity_lib_dir():
+    import jnius  # should get us our fake module
+
+    # Short test that it works when activity doesn't exist:
+    jnius.autoclass = MagicMock()
+    jnius.autoclass.return_value = None
+    assert android._ctypes_library_finder.get_activity_lib_dir(
+        "JavaClass"
+    ) is None
+    assert mock.call("JavaClass") in jnius.autoclass.call_args_list
+
+    # Comprehensive test that verifies getApplicationInfo() call:
+    activity = MagicMock()
+    app_context = activity.getApplicationContext()
+    app_context.getPackageName.return_value = "test.package"
+    app_info = app_context.getPackageManager().getApplicationInfo()
+    app_info.nativeLibraryDir = '/testpath'
+
+    def pick_class(name):
+        cls = MagicMock()
+        if name == "JavaClass":
+            cls.mActivity = activity
+        elif name == "android.content.pm.PackageManager":
+            # Manager class:
+            cls.GET_SHARED_LIBRARY_FILES = 1024
+        return cls
+
+    jnius.autoclass = MagicMock(side_effect=pick_class)
+    assert android._ctypes_library_finder.get_activity_lib_dir(
+        "JavaClass"
+    ) == "/testpath"
+    assert mock.call("JavaClass") in jnius.autoclass.call_args_list
+    assert mock.call("test.package", 1024) in (
+        app_context.getPackageManager().getApplicationInfo.call_args_list
+    )
+
+
+def test_find_library():
+    test_d = tempfile.mkdtemp(prefix="p4a-android-ctypes-test-libdir-")
```

Comment:
I was glad to see we actually have context manager for this. If you use it we can ditch the try/finally.
```
with tempfile.TemporaryDirectory(prefix='p4a-android-ctypes-test-libdir-') as test_d:
```

Reply:
@AndreMiras very nice find! sadly it appears to be python 3 only, I just tested. So maybe something to leave for our more later future when we go python 2 only?

Reply:
OK thanks I didn't realise, let's leave it as you have then :+1: